lib/utils.bash

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+# Copyright (C) 2019-2020 Jorge Canha
+#
+# This file is part of asdf-pulumi.
+#
+# asdf-pulumi is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# asdf-pulumi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with asdf-pulumi.  If not, see <http://www.gnu.org/licenses/>.
+
+set -euo pipefail
+
+GH_REPO="https://github.com/pulumi/pulumi"
+TOOL_NAME="pulumi"
+TOOL_TEST="pulumi version"
+
+get_platform() {
+  local uname_platform="$(uname)"
+
+  case "${uname_platform}" in
+  "Darwin")
+    echo "darwin"
+    ;;
+
+  "Linux")
+    echo "linux"
+    ;;
+
+  *)
+    echo "Platform ${uname_platform} is not supported."
+    exit 1
+    ;;
+  esac
+}
+
+get_architecture() {
+  local uname_arch="$(uname -m)"
+
+  case "${uname_arch}" in
+  "arm64" | "aarch64")  # Mac M1 and Linux ARM
+    echo "arm64"
+    ;;
+
+  "x86_64")
+    echo "x86"
```

Comment:
Started running a few tests on this PR, this string actually doesn't work for amd64 cpu's, please change it to x64:

`Updating pulumi to jans-forks/master
Already on 'jans-forks/master'
Your branch is up to date with 'origin/jans-forks/master'.
* Downloading pulumi release 3.40.2...
curl: (22) The requested URL returned error: 404
asdf-pulumi: Could not download https://github.com/pulumi/pulumi/releases/download/v3.40.2/pulumi-v3.40.2-linux-x86.tar.gz
FAILED: install exited with an error
`

Reply:
Ping @1oglop1

Reply:
Changed

Reply:
I was already preparing a branch to run some tests on CircleCI, I should publish it soon, let's first merge this PR.